src/components/LocationPage/ShareButtons.js

```diff
@@ -0,0 +1,74 @@
+import React, { Fragment, useState } from 'react';
+import SocialButtons from './SocialButtons';
+import {
+  SaveOrShareContainer,
+  SaveOrShareButton,
+  DesktopButtonsWrapper,
+  MobileButtonsWrapper,
+} from './ShareButtons.style';
+import { useTheme } from '@material-ui/core/styles';
+import { ClickAwayListener, useMediaQuery } from '@material-ui/core';
+
+const InnerContent = props => {
+  const { iconSize, setShowShareIcons, showShareIcons } = props;
+
+  return (
+    <Fragment>
+      <SaveOrShareContainer>
+        <SaveOrShareButton
+          onClick={() => {
+            setShowShareIcons(false);
+          }}
+        >
+          Save
+        </SaveOrShareButton>
+        <SaveOrShareButton
+          isLast
+          onClick={() => {
+            setShowShareIcons(!showShareIcons);
+          }}
+        >
+          Share
+        </SaveOrShareButton>
+      </SaveOrShareContainer>
+      {showShareIcons && <SocialButtons iconSize={iconSize} />}
+    </Fragment>
+  );
+};
+
+const ShareButtons = props => {
+  const { isFirst } = props;
+  const [showShareIcons, setShowShareIcons] = useState(false);
+
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('xs'));
+
+  return (
+    <Fragment>
+      {isMobile && (
+        <ClickAwayListener onClickAway={() => setShowShareIcons(false)}>
```

Comment:
Rather than duplicate the `ClickAwayListener` for both `isMobile` and `!isMobile` could you move it outside (i.e. right below `<Fragment>` ?

Alternatively would it work to put this inside `InnerContent`?  And then you could move `showShareIcons` state 100% inside of `InnerContent`.  Not sure if that'll cause some issue though.